src/cli/study-backrefs.js

```diff
@@ -0,0 +1,358 @@
+];
+
+const report = {};
+
+function recordAnomaly(spec, anomalyType, link) {
+  if (!report[spec.url]) {
+    report[spec.url] = {
+      title: spec.title,
+      notExported: [],
+      notDfn: [],
+      brokenLink: [],
+      evolvingLink: [],
+      outdatedSpec: [],
+      unknownSpec: [],
+      datedUrl: []
+    };
+  }
+  report[spec.url][anomalyType].push(link);
+}
+
+const missingSpecs = {};
+
+function recordUnknownSpec(link, shortname, spec) {
+  if (!missingSpecs[shortname]) {
+    missingSpecs[shortname] = [];
+  }
+  missingSpecs[shortname].push({link, spec: spec.url});
+}
+
+function studyCrawlResults(edResults, trResults) {
+  edResults.forEach(spec => {
+    Object.keys(spec.links)
+      .filter(matchSpecUrl)
+      .forEach(l => {
+        let shortname;
+        let nakedLink = l.replace(/#.*$/, '');
+        if (nakedLink.endsWith(".html")) {
+          nakedLink = nakedLink.replace(/\/[^/]*\.html/, '/');
+        }
+        if (nakedLink[nakedLink.length - 1] !== '/') {
+          nakedLink += '/';
+        }
+        shortname = (edResults.find(r => r.url === nakedLink || (r.release && r.release.url === nakedLink) || r.nightly.url === nakedLink || (r.series && nakedLink === 'https://www.w3.org/TR/' + r.series.shortname + '/') ) || {}).shortname;
+        if (!shortname) {
+          try {
+            ({shortname} = computeShortname(l));
+          } catch (e) {
+            let m = l.match(/www\.w3\.org\/TR\/[0-9]{4}\/[A-Z]+-(.+)-[0-9]{8}/);
+            if (m) {
+              // Ignore links to previous versions
+              if (m[1] !== spec.shortname) {
+                recordAnomaly(spec, "datedUrl", l);
+              }
+            } else {
+              recordAnomaly(spec, "unknownSpec", l);
+            }
+          }
+        }
+        if (shortnameMap[shortname]) {
+          shortname = shortnameMap[shortname];
+        } else if (shortNamesOfOutdatedSpecs[shortname]) {
+          recordAnomaly(spec, "outdatedSpec", l);
+        }
+        if (!shortname) { return ;}
+        // self-references might be broken because of ed vs tr
+        if (shortname === spec.shortname || shortname === spec.series.shortname) return [];
+        let sourceSpec = edResults.find(s => s.shortname === shortname || s.series.shortname === shortname);
+        let trSourceSpec = trResults.find(s => s.shortname === shortname || s.series.shortname === shortname);
+        if (!sourceSpec) {
+          if (shortname && !shortNamesOfOutdatedSpecs[shortname] && !shortnameOfNonNormativeDocs.includes(shortname)) {
+            recordUnknownSpec(l, shortname, spec)
+          }
+          return;
+        }
+        let headings = sourceSpec.headings || [];
+        let dfns = sourceSpec.dfns || [];
+        let ids = sourceSpec.ids || [];
+
+        // anchors
+        const anchors = spec.links[l];
+        for(let anchor of anchors) {
+          let id = ids.includes(anchor);
```

Comment:
Calling a boolean `id` is a good way to obfuscate the code...

Reply:
always happy to help